src/httpClient.ts

```diff
@@ -151,3 +152,113 @@ export class HttpClientResponse<T> implements IHttpClientResponse {
         this.body = body;
     }
 }
+
+export class HttpApiClient extends HttpClient {
+
+    baseUrl: any;
+
+    constructor(baseUrl: string = undefined) {
+        super();
+        this.baseUrl = baseUrl;
+    }
+
+    request(method: string, url: string, data: any, files: any[], options: IHttpClientRequestOptions) {
+
+        if (typeof method !== "string") {
+            throw new Error("method must be a string");
+        }
+
+        if (["GET", "HEAD", "DELETE", "POST", "PATCH", "PUT"].indexOf(method) < 0) {
+            throw new Error("method can be one of GET, HEAD, DELETE, PATCH, POST, PUT");
+        }
+
+        if (typeof url !== "string") {
+            throw new Error("url must be a string");
+        }
+
+        if (typeof data !== "object" && typeof data !== "undefined") {
+            throw new Error("data must be an object or undefined");
+        }
+
+        if (typeof files !== "object" && typeof files !== "undefined") {
+            throw new Error("files must be an object or undefined");
+        }
+
+        if (typeof options !== "object" && typeof options !== "undefined") {
+            throw new Error("options must be an object or undefined");
+        }
+
+        // Validate method/parameter combinations
+        switch (method) {
+            case "GET":
+                if (data || files) {
+                    throw new Error("data and files cannot be defined on a GET call");
+                }
+                break;
+            case "HEAD":
+                if (data || files) {
+                    throw new Error("data and files cannot be defined on a HEAD call");
+                }
+                break;
+            case "DELETE":
+                if (files) {
+                    throw new Error("files cannot be defined on a DELETE call");
+                }
+                break;
+        }
+
+        options = options || {};
+
+        options.method = method;
+        options.url = url;
+
+        if (data) {
+            options.data = data;
```

Comment:
logic problem. data can be all kinds of falsy values and still be correct! E.g. `false`, `null` or `0` are perfectly valid values which will not be passed on to data. It's best to just accept data as it is (because it can be undefined or virtually anything ^^)

Update your tests and implementation for this :)